src/libs/ui/widgets/treeview.cpp

```diff
@@ -0,0 +1,44 @@
+/****************************************************************************
+**
+** Copyright (C) 2015-2016 Oleg Shparber
+** Copyright (C) 2013-2014 Jerzy Kozera
+** Contact: https://go.zealdocs.org/l/contact
+**
+** This file is part of Zeal.
+**
+** Zeal is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** Zeal is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Zeal. If not, see <https://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include "treeview.h"
+
+using namespace Zeal::WidgetUi;
+
+TreeView::TreeView(QWidget *parent) :
+    QTreeView(parent)
+{
+}
+
+void TreeView::selectionChanged(const QItemSelection &selected, const QItemSelection &deselected)
+{
+    QTreeView::selectionChanged(selected, deselected);
```

Comment:
There's no need in this `TreeView` class. Selection updates can be obtained via [`QAbstractitemView::selectionModel`](http://doc.qt.io/qt-5/qabstractitemview.html#selectionModel).

Reply:
See the comment below that - it's the first approach I tried, but it doesn't work reliably:

> `QTreeView` doesn't normally expose selection changes as a public signal. `QTreeView's`
> selection model is accessible publicly, however the instance this refers to can change
> internally, so it's not reliable to connect these `QItemSelectionModel` signals externally.

The selection model gets switched out regularly in the current implementation, so you end up with a pointer to an old selection model after the first search and the connection no longer reports selection changes.